jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/PersistenceUnitServiceHandler.java

```diff
@@ -442,6 +452,41 @@ private static void deployPersistenceUnit(
         }
     }
 
+    private static void handleClassChange(PersistenceUnitServiceImpl service, DeploymentClassChangeSupport classChangeSupport) {
+        if(classChangeSupport == null) {
+            return;
+        }
+        classChangeSupport.addListener(new ClassChangeListener() {
+            @Override
+            public void classesReplaced(List<ChangedClasssDefinition> replacedClasses, List<NewClassDefinition> newClassDefinitions) {
```

Comment:
what should the ordering sequence be for:
- Updated class definitions being loaded by classloader.
- classReplaced() being called.
- Weld rewriting application classes.
- Persistence provider [rewriting application (entity) classes](https://docs.oracle.com/javaee/7/api/javax/persistence/spi/ClassTransformer.html).